### ISSUE TITLE
Improve train/test split

### ISSUE THREAD

**Issue by mastoffel** [alan-turing-institute/autoemulate#248]
There's a lot to improve in our training / test set split in `data_splitting.py`:
* the data is always split. However, we would want `test_set_size=0` to work, so there's no split and `AutoEmulate` uses the full data as training data. We then have to decide what to do with `evaluate`, as there's no test set to evaluate on. We could allow it to take an external test set, otherwise throw an error. Similarly, `refit` should throw an error, as the models are refitted on the full 'training' data after cross-validation already.
* data is randomly shuffled before being split. For temporal data, spatial data, grouped data etc. we want more sophisticated splitting techniques.
